### ISSUE TITLE
latest drone can't activate repo (can't install hook)

### ISSUE THREAD

**Issue by vtolstov** [go-gitea/gitea#156]
gitea have in logs
```
Jul 26 01:02:06 selfip.ru gitea[28864]: [Macaron] Started POST /api/v1/repos/vtolstov/cb-mdadm/hooks for 37.139.40.30
Jul 26 01:02:06 selfip.ru gitea[28864]: [Macaron] Completed /api/v1/repos/vtolstov/cb-mdadm/hooks 422  in 8.844291ms
```

and nothing in drone happening

**Comment by tboerger** [comment]
```422``` seems to be the error code, unprocessable entity. So there must be some debug information.

**Comment by vtolstov** [comment]
I'm try enable dev/test mode, Trace mode in logs, but no other messages displayed

**Comment by vtolstov** [comment]
I'm switch from console to file log and get some more info when i click activate:
```
[xorm] [info]  2015/07/26 01:32:06.719840 [sql] SELECT `id`, `uid`, `name`, `sha1`, `created`, `updated` FROM `access_token` WHERE `sha1` = ? LIMIT 1 [args] [b2e730a7f062e162c4ab82abbc9a74362019c2e0]
[xorm] [info]  2015/07/26 01:32:06.721251 [sql] SELECT `id`, `lower_name`, `name`, `full_name`, `email`, `passwd`, `login_type`, `login_source`, `login_name`, `type`, `location`, `website`, `rands`, `salt`, `created`, `updated`, `is_active`, `is_admin`, `allow_git_hook`, `avatar`, `avatar_email`, `use_custom_avatar`, `num_followers`, `num_followings`, `num_stars`, `num_repos`, `description`, `num_teams`, `num_members` FROM `user` WHERE `id` = ? LIMIT 1 [args] [2]
[xorm] [info]  2015/07/26 01:32:06.722968 [sql] SELECT `id`, `owner_id`, `lower_name`, `name`, `description`, `website`, `default_branch`, `num_watches`, `num_stars`, `num_forks`, `num_issues`, `num_closed_issues`, `num_pulls`, `num_closed_pulls`, `num_milestones`, `num_closed_milestones`, `is_private`, `is_bare`, `is_mirror`, `is_fork`,
`fork_id`, `is_wiki`, `wiki_repo_id`, `created`, `updated` FROM `repository` WHERE `owner_id` = ? AND `lower_name` = ? LIMIT 1 [args] [2 cb-mdadm]
[xorm] [info]  2015/07/26 01:32:06.724653 [sql] SELECT `id`, `lower_name`, `name`, `full_name`, `email`, `passwd`, `login_type`, `login_source`, `login_name`, `type`, `location`, `website`, `rands`, `salt`, `created`, `updated`, `is_active`, `is_admin`, `allow_git_hook`, `avatar`, `avatar_email`, `use_custom_avatar`, `num_followers`, `num_followings`, `num_stars`, `num_repos`, `description`, `num_teams`, `num_members` FROM `user` WHERE `id` = ? LIMIT 1 [args] [2]
```

**Comment by vtolstov** [comment]
tcpdump displays:
```
..A...>.HTTP/1.0 500 Internal Server Error
Access-Control-Allow-Origin: *
Cache-Control: no-cache
Cache-Control: no-store
Cache-Control: max-age=0
Cache-Control: must-revalidate
Cache-Control: value
Content-Type: text/plain; charset=utf-8
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Last-Modified: Sat, 25 Jul 2015 22:33:30 GMT
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-Xss-Protection: 1; mode=block
Date: Sat, 25 Jul 2015 22:33:31 GMT
Content-Length: 18

invalid hook type
```

**Comment by tboerger** [comment]
@lunny any idea why it results in 422?

**Comment by vtolstov** [comment]
hah i found problem. drone use hook type gogs, but gitea change hook name from gogs to gitea:

```
var hookTaskTypes = map[string]HookTaskType{
▶       "gitea":  GITEA,
▶       "slack": SLACK,
}

// ToHookTaskType returns HookTaskType by given name.
func ToHookTaskType(name string) HookTaskType {
▶       return hookTaskTypes[name]
}

func (t HookTaskType) Name() string {
▶       switch t {
▶       case GITEA:
▶       ▶       return "gitea"
▶       case SLACK:
▶       ▶       return "slack"
▶       }
▶       return ""
}
```

**Comment by vtolstov** [comment]
@tboerger may be add gogs hook time and check gitea and gogs in the same way?

**Comment by tboerger** [comment]
AHHH, that's because we already renamed the hook tpye to gitea, drone uses gogs :)

**Comment by vtolstov** [comment]
@tboerger can you create pr ?

**Comment by tboerger** [comment]
maybe we need to fallback to gogs until drone implements gitea.

**Comment by tboerger** [comment]
If you have not done it until tomorrow I won't do it for the next week as I will go on vacation for a week.

**Comment by vtolstov** [comment]
may be pr #161 fixes this ?=)

**Comment by tboerger** [comment]
Fixed with #161
